openquake/hazardlib/gsim/manea_2021.py

```diff
@@ -0,0 +1,298 @@
+    """
+    ssa, ssb, ssc = _get_site_type_dummy_variables(sites)
+
+    return ((ssc*C['phi11'] + ssb*C['phi10'] + ssa*C['phi9']) +
+            _compute_site_amplif(C, sites))
+
+
+def _get_site_type_dummy_variables(sites):
+    """
+    Get site type dummy variables, three different EC8 site classes
+    The recording sites are classified into 3 classes,
+    based on the shear wave velocity intervals in the uppermost 30 m, Vs30,
+    according to the EC8 (CEN 2003):
+
+    class A: Vs30 > 800 m/s
+    class B: Vs30 = 360 − 800 m/s
+    class C: Vs30 = 180 - 360 m/s
+    """
+    ssa = np.zeros(len(sites.vs30))
+    ssb = np.zeros(len(sites.vs30))
+    ssc = np.zeros(len(sites.vs30))
+    # SClass C; 180 m/s <= Vs30 <= 360 m/s.
+    idx = (sites.vs30 >= 180.0) & (sites.vs30 < 360.0)
+    ssc[idx] = 1.0
+    # Class B; 360 m/s <= Vs30 <= 800 m/s.
+    idx = (sites.vs30 >= 360.0) & (sites.vs30 < 800)
+    ssb[idx] = 1.0
+    # Class A; Vs30 > 800 m/s.
+    idx = (sites.vs30 >= 800.0)
+    ssa[idx] = 1.0
+    return ssa, ssb, ssc
+
+
+def _compute_site_amplif(C, sites):
+    """
+    Compute site amplification using fundamental frequency of resonance(f0)
+    If f0 is not defined, please set it equal with the reference frequency
+    (fref=15, attributed to the reference site condition-rock sites)
+    """
+    return C['phi8'] * np.log(sites.f0 / 15)
+
+
+def get_mean_values(C, ctx, arc):
+    """
+    Returns the mean values for a specific IMT
+    """
+    mean = (_compute_magnitude_term(C, ctx.mag) +
+            _compute_distance_term(C, ctx.rhypo, arc) +
+            _compute_depth_term(C, ctx.hypo_depth) +
+            _get_site_amplification(C, ctx))
+    return mean
+
+
+class ManeaEtAl2021fore(GMPE):
+    """
+    Implements the Subduction GMPE developed by Elena Florinela Manea,
+    Carmen Ortanza Cioflan and Laurentiu Danciu, otherwise known as the
+    "Ground-motion models for Vrancea intermediate-depth earthquakes
+    (Earthquake Spectra,2021,87552930211032985), for subduction inslab events.
+    This class is defined for sites located in the gfrearc area
+    https://journals.sagepub.com/doi/abs/10.1177/87552930211032985
+    """
+    DEFINED_FOR_TECTONIC_REGION_TYPE = const.TRT.SUBDUCTION_INTRASLAB
+
+    DEFINED_FOR_INTENSITY_MEASURE_TYPES = {PGA, SA}
+
+    DEFINED_FOR_INTENSITY_MEASURE_COMPONENT = const.IMC.AVERAGE_HORIZONTAL
+
+    DEFINED_FOR_STANDARD_DEVIATION_TYPES = {
+        const.StdDev.TOTAL, const.StdDev.INTER_EVENT, const.StdDev.INTRA_EVENT}
+
+    #: Required site measure as Vs30, fundamental frequency of resonance
+    #: (if unknown set it as 15 - rock sites),
+    REQUIRES_SITES_PARAMETERS = {'vs30', 'f0'}
+
+    #: Required rupture parameters are magnitude and depth for the inslab model
+    REQUIRES_RUPTURE_PARAMETERS = {'mag', 'hypo_depth'}
+
+    #: Required distance measure is hypocentral distance
+    REQUIRES_DISTANCES = {'rhypo'}
+
+    #: arc set to 2 for fore - arc sites
```

Comment:
I have some doubts about this. I would rather take the backarc / forearc from the site properties and avoid having separate models for the forearc and the backarc. It would be extremely difficult to apply properly the model as is in a hazard analysis that includes several sites with different position (and therefore fore-arc and back-arc properties).

See for example https://github.com/gem/oq-engine/blob/master/openquake/hazardlib/gsim/abrahamson_2015.py#L278 and https://github.com/gem/oq-engine/blob/master/openquake/hazardlib/gsim/abrahamson_2015.py#L120